lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2233,90 +2253,60 @@ namespace {
+                tryMarkImplicitlyThrows(memberLoc, memberRef, context);
             } else {
-              // the func is neither static or distributed
-              ctx.Diags.diagnose(memberLoc, diag::distributed_actor_isolated_method);
-              // TODO: offer a fixit to add 'distributed' on the member; how to test fixits? See also https://github.com/apple/swift/pull/35930/files
+              // neither static or distributed, apply full distributed isolation
+              ctx.Diags.diagnose(memberLoc, diag::distributed_actor_isolated_method)
+                  .fixItInsert(member->getAttributeInsertionLoc(true), "distributed ");
               noteIsolatedActorMember(member, context);
               return true;
             }
           } // end FuncDecl
 
-          if (!continueToCheckingLocalIsolation) {
-            // it wasn't a function (including a distributed function),
-            // so we need to perform some more checks
-            if (auto var = dyn_cast<VarDecl>(member)) {
-              // TODO: we want to remove _distributedActorIndependent in favor of nonisolated
-              //
-              // @_distributedActorIndependent decls are accessible always,
-              // regardless of distributed actor-isolation; e.g. actorAddress
-              if (member->getAttrs().hasAttribute<DistributedActorIndependentAttr>())
-                return false;
-
-              // nonisolated decls are accessible always
-              if (member->getAttrs().hasAttribute<DistributedActorIndependentAttr>())
-                return false;
-
-              // otherwise, no other properties are accessible on a distributed actor
-              if (!continueToCheckingLocalIsolation) {
-                ctx.Diags.diagnose(
-                    memberLoc, diag::distributed_actor_isolated_non_self_reference,
-                    member->getDescriptiveKind(),
-                    member->getName());
-                noteIsolatedActorMember(member, context);
-                return true;
-              }
-            } // end VarDecl
+          if (isPropOrSubscript(member)) {
+            DeclAttributes &attrs = member->getAttrs();
+            if (attrs.hasAttribute<DistributedActorIndependentAttr>())
+              return false;
 
-            // TODO: would have to also consider subscripts and other things
-          }
-        } // end !isolatedActor
+            if (attrs.hasAttribute<NonisolatedAttr>())
+              return false;
 
-        if (!continueToCheckingLocalIsolation)
-          return false;
+              ctx.Diags.diagnose(
+                  memberLoc, diag::distributed_actor_isolated_non_self_reference,
+                  member->getDescriptiveKind(),
+                  member->getName());
+              noteIsolatedActorMember(member, context);
+              return true;
+          } // end VarDecl
+        }
 
-        LLVM_FALLTHROUGH;
+        return diagnoseNonSendableTypesInReference(
```

Comment:
This brings an interesting question to mind. If we're going through the distributed entry point, do we even need `Sendable` checking? Or is that only needed in the local case?

Reply:
I think it still matters -- the transport may use a different thread to serialize things (one of my impls does so), so the thread safety of the payloads is still useful

Reply:
Even through I guess the only thing it does to them is invoke encode... 🤔